runtime/lua/vim/_defaults.lua

```diff
@@ -190,7 +190,10 @@ do
     })
 
     vim.keymap.set('n', '<C-W>d', function()
-      vim.diagnostic.open_float({ border = 'rounded' })
+      local config = vim.diagnostic.config()
+      local border = type(config.float) == 'table' and (config.float.border or 'rounded')
+        or 'rounded'
```

Comment:
could use `vim.tbl_get`.  

side-note:  should `vim.tbl_get` allow non-table input? that would make it more flexible, but less strict.

```suggestion
      local border = vim.tbl_get(config, 'float', 'border') or 'rounded'
```

Reply:
forget some util functions someitme .......